cumulus/ansible/tasks/cluster.py

```diff
@@ -155,22 +155,25 @@ def start_cluster(launch_playbook, provision_playbook, cluster, profile,
         aws_secret_access_key=secret_key,
         cluster_id=cluster['_id'])
 
-    master = parse('head.hosts[0]').find(inventory)
-    if master:
-        master = master[0].value
-    else:
-        raise Exception('Unable to extract cluster master host.')
-
-    status_url = '%s/clusters/%s' % (cumulus.config.girder.baseUrl,
-                                     cluster['_id'])
-    updates = {
-        'config': {
-            'host': master
+    # If master_name has been provide extract out the master host and
+    # update the cluster with it.
+    if master_name:
+        master = parse('%s.hosts[0]' % master_name).find(inventory)
+        if master:
+            master = master[0].value
+        else:
+            raise Exception('Unable to extract cluster master host.')
+
+        status_url = '%s/clusters/%s' % (cumulus.config.girder.baseUrl,
+                                         cluster['_id'])
+        updates = {
+            'config': {
+                'host': master
+            }
```

Comment:
The config.host property is currently used to create an ssh connection to the head node. This is not used exclusively for job submission. For example its used for access to any shared file systems the head node might be sharing.